File: study/lua/src/app/scene/view/tenJadeAuction/TenJadeAuctionConfigHelper.lua
local TenJadeAuctionConfigHelper = {}
local TenJadeAuctionPrice = require("app.config.ten_jade_auction_price")
local TenJadeAuctionConst  = require("app.const.TenJadeAuctionConst")

------------------------------------------------------------
--------------ten_jade_auction_price 配置-----------------------
------------------------------------------------------------
function TenJadeAuctionConfigHelper.getPriceConfig(id)
    local info = require("app.config.ten_jade_auction_price").get(id)
	assert(info, string.format("ten_jade_auction_price config can not find id = %d", id))
	return info
end

------------------------------------------------------------
--------------ten_jade_auction_para 配置--------------------
------------------------------------------------------------
--拍卖前开启倒计时（秒）
function TenJadeAuctionConfigHelper.getCountDown()
    local info = require("app.config.ten_jade_auction_para").get(TenJadeAuctionConst.OPEN_COUNT_DOWN)
    assert(info, string.format("ten_jade_auction_para config can not find id = %d", TenJadeAuctionConst.OPEN_COUNT_DOWN))
    return tonumber(info.content)
end

--顶价加时触发时间（拍卖结束前x秒内顶价，会增加结束时间）
function TenJadeAuctionConfigHelper.getExtendStartTime()
    local info = require("app.config.ten_jade_auction_para").get(TenJadeAuctionConst.EXTEND_START_TIME)
    assert(info, string.format("ten_jade_auction_para config can not find id = %d", TenJadeAuctionConst.EXTEND_START_TIME))
    return tonumber(info.content)
end

--顶价加时（秒）
function TenJadeAuctionConfigHelper.getExtendTime()
    local info = require("app.config.ten_jade_auction_para").get(TenJadeAuctionConst.EXTEND_TIME)
    assert(info, string.format("ten_jade_auction_para config can not find id = %d", TenJadeAuctionConst.EXTEND_TIME))
    return tonumber(info.content)
end

--玩家可以看到活动入口和广告页的时间（活动当天x时x分x秒）
function TenJadeAuctionConfigHelper.getEntranceOpenTime()
    local info = require("app.config.ten_jade_auction_para").get(TenJadeAuctionConst.ENTRANCE_OPEN_TIME)
    assert(info, string.format("ten_jade_auction_para config can not find id = %d", TenJadeAuctionConst.ENTRANCE_OPEN_TIME))
    local array = string.split(info.content, "|")
    return tonumber(array[1]), tonumber(array[2]), tonumber(array[3])
end

--限时活动展示道具id
function TenJadeAuctionConfigHelper.getShowItem(index)
    local id = TenJadeAuctionConst.SHOW_ICON_START + index - 1
    local info = require("app.config.ten_jade_auction_para").get(id)
    assert(info, string.format("ten_jade_auction_para config can not find id = %d", id))
    local array = string.split(info.content, "|")
    return tonumber(array[1]), tonumber(array[2])
end


------------------------------------------------------------
--------------------------方法-------------------------------
------------------------------------------------------------
--根据竞拍人数确定跳多少价格
function TenJadeAuctionConfigHelper.getPriceAddWithIndex()
    local info = TenJadeAuctionConfigHelper.getPriceConfig(id)
    return info.price_add
end

--活动入口显示时间戳
function TenJadeAuctionConfigHelper.getAuctionOpenTimeStamp()
    local openTime = G_UserData:getTenJadeAuction():getCurAuctionStartTime()
    local h, m, s = TenJadeAuctionConfigHelper.getEntranceOpenTime()
    local openTimeZero = G_ServerTime:secondsFromZero(openTime)
    return openTimeZero + h * 3600 + m * 60 + s
end

--处于哪个阶段
function TenJadeAuctionConfigHelper.getAuctionPhase()
    local curTime = G_ServerTime:getTime()
    local curAuctionInfo = G_UserData:getTenJadeAuction():getCurAuctionInfo()
    if not G_UserData:getTenJadeAuction():hasAuction() then
        return TenJadeAuctionConst.PHASE_END
    end
    local endTime = curAuctionInfo:getEnd_time()
    local startTime = curAuctionInfo:getStart_time()
    local countDown = TenJadeAuctionConfigHelper.getCountDown()
    if curTime < TenJadeAuctionConfigHelper.getAuctionOpenTimeStamp() then
        return TenJadeAuctionConst.PHASE_DEFAULT
    elseif curTime < startTime then
        return TenJadeAuctionConst.PHASE_SHOW
    elseif curTime < startTime + countDown then
        return TenJadeAuctionConst.PHASE_ITEM_SHOW
    elseif curTime < endTime then
        return TenJadeAuctionConst.PHASE_START
    else
        return TenJadeAuctionConst.PHASE_END
    end
end



return TenJadeAuctionConfigHelper

File: study/ts/script/scene/view/tenJadeAuction/TenJadeAuctionConfigHelper.ts
import { ConfigNameConst } from "../../../const/ConfigNameConst";
import { TenJadeAuctionConst } from "../../../const/TenJadeAuctionConst";
import { G_ConfigLoader, G_ServerTime, G_UserData } from "../../../init";
import { assert } from "../../../utils/GlobleFunc";


export class TenJadeAuctionConfigHelper{
static  getPriceConfig(id) {
    var info = G_ConfigLoader.getConfig(ConfigNameConst.TEN_JADE_AUCTION_PRICE).get(id);
    assert(info, 'ten_jade_auction_price config can not find id = %d');
    return info;
};
static  getCountDown() {
    var info = G_ConfigLoader.getConfig(ConfigNameConst.TEN_JADE_AUCTION_PARA).get(TenJadeAuctionConst.OPEN_COUNT_DOWN);
    assert(info, 'ten_jade_auction_para config can not find id = %d');
    return parseInt(info.content);
};
static  getExtendStartTime() {
    var info = G_ConfigLoader.getConfig(ConfigNameConst.TEN_JADE_AUCTION_PARA).get(TenJadeAuctionConst.EXTEND_START_TIME);
    assert(info, 'ten_jade_auction_para config can not find id = %d');
    return parseInt(info.content);
};
static  getExtendTime() {
    var info = G_ConfigLoader.getConfig(ConfigNameConst.TEN_JADE_AUCTION_PARA).get(TenJadeAuctionConst.EXTEND_TIME);
    assert(info, 'ten_jade_auction_para config can not find id = %d');
    return parseInt(info.content);
};
static  getEntranceOpenTime() {
    var info = G_ConfigLoader.getConfig(ConfigNameConst.TEN_JADE_AUCTION_PARA).get(TenJadeAuctionConst.ENTRANCE_OPEN_TIME);
    assert(info, 'ten_jade_auction_para config can not find id = %d');
    var array = info.content.split('|');
    return [
        parseInt(array[0]),
        parseInt(array[1]),
        parseInt(array[2])
    ];
};
static  getShowItem(index) {
    var id = TenJadeAuctionConst.SHOW_ICON_START + index - 1;
    var info = G_ConfigLoader.getConfig(ConfigNameConst.TEN_JADE_AUCTION_PARA).get(id);
    assert(info, 'ten_jade_auction_para config can not find id = %d');
    var array = info.content.split('|');
    return [
        parseInt(array[0]),
        parseInt(array[1])
    ];
};
static  getPriceAddWithIndex() {
    // var info = TenJadeAuctionConfigHelper.getPriceConfig(id);
    // return info.price_add;
};
static  getAuctionOpenTimeStamp() {
    var openTime = G_UserData.getTenJadeAuction().getCurAuctionStartTime();
    var [h, m, s] = TenJadeAuctionConfigHelper.getEntranceOpenTime();
    var openTimeZero = G_ServerTime.secondsFromZero(openTime);
    return openTimeZero + h * 3600 + m * 60 + s;
};
static getAuctionPhase () {
    var curTime = G_ServerTime.getTime();
    var curAuctionInfo = G_UserData.getTenJadeAuction().getCurAuctionInfo();
    if (!G_UserData.getTenJadeAuction().hasAuction()) {
        return TenJadeAuctionConst.PHASE_END;
    }
    var endTime = curAuctionInfo.getEnd_time();
    var startTime = curAuctionInfo.getStart_time();
    var countDown = TenJadeAuctionConfigHelper.getCountDown();
    if (curTime < TenJadeAuctionConfigHelper.getAuctionOpenTimeStamp()) {
        return TenJadeAuctionConst.PHASE_DEFAULT;
    } else if (curTime < startTime) {
        return TenJadeAuctionConst.PHASE_SHOW;
    } else if (curTime < startTime + countDown) {
        return TenJadeAuctionConst.PHASE_ITEM_SHOW;
    } else if (curTime < endTime) {
        return TenJadeAuctionConst.PHASE_START;
    } else {
        return TenJadeAuctionConst.PHASE_END;
    }
};
}